### PR TITLE
Fix: fee recovery in the “donation summary” with wrong value

### LINKED ISSUE #6536: The donation summary should correctly display the fee recovery amount for currencies that uses a comma as a decimal separator
## User Story

When using the fee recovery amount add-on with a currency that uses a comma as a decimal separator,  a wrong fee value is displayed in the "donation summary" section.

## Details

If you check the "cover transaction cost" option and get €3,30 as the fee for example. Then when you go to the next page, in the "donation summary" section, you'll see €33,00 instead of the original value.

## Expected Behavior

You should see the same value from the "cover transaction cost" option in the "donation summary" section.

## Steps to Reproduce

1. Install and active the Fee Recovery Add-on
2. Setup a multi-step or classic form template
3. Use a currency that uses a comma as a decimal separator like Euro (€)
4. Make a test donation and tick the "cover transaction cost" option
5. Before finishing the donation, on the "donation summary" section you should see the wrong value for the "donation fees" value

## Visuals

https://www.loom.com/share/bb5c72f5060343eb8bcdc3b7a46a6fba

## Acceptance Criteria

-   [x] The fee value in the “donation summary” accurately reflects the fee recovery amount.

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves #6536

## Description

The donation summary is displaying the wrong fee recovery amount for currencies that uses a comma as a decimal separator. 

~~This PR solves the problem by adding extra param (with "true" as the default value) to the `format_amount()` function that allows us to set if the amount should be normalized or not because when it's called inside the `handleFees()` function the value passed through is already normalized.~~

~~So, when the value is already normalized, we need to set this extra param to "false" to prevent the double normalization process - because when it happens, a wrong value is generated.~~

This PR solves the problem by getting the formatted value directly from the DOM instead of retrieving it from data attributes and using the `format_amount()` function to normalize these values - in fact, this function is no longer useful and has been removed.

Another problem solved by this PR is that now the fee amount gets updated even when the `GiveWP > Settings > Fee Recovery > Fee Opt-In > Forced Opt-in` is enabled - this option removes the checkbox observed on the frontend and it also was preventing the fee amount update. 

Now the frontend no more observes the checkbox to handle the fee amount changes, instead that the `handleFees()` function is triggered every time that the `.give-final-total-amount` element gets updated. This way, some old codes responsible for updating the fee amount were removed from some files because they are no more necessary.

## Visuals

Before fix: https://youtu.be/PcwG0vrvybU

After fix: https://youtu.be/jDG-YHV1ru8

## Testing Instructions

1. Install and active the Fee Recovery Add-on
2. Setup a multi-step or classic form template
3. Use a currency that uses a comma as a decimal separator like Euro (€)
4. Make a test donation and tick the "cover transaction cost" option
5. Before finishing the donation, the fee value in the “donation summary” should accurately reflect the fee recovery amount
6. Also make a second test donation with the `GiveWP > Settings > Fee Recovery > Fee Opt-In > Forced Opt-in` enabled

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/DonationSummary/resources/js/summary.js [hunk 1]
```diff
@@ -205,12 +205,15 @@ window.GiveDonationSummary = {
...
-    format_amount: function (amount, $form) {
+    format_amount: function (amount, $form, normalize = true) {
         // Normalize amounts to JS number format
-        amount = amount
-            .replace(Give.form.fn.getInfo('thousands_separator', $form), '')
-            .replace(Give.form.fn.getInfo('decimal_separator', $form), '.');
+        if (normalize) {
+            amount = amount
+                .replace(Give.form.fn.getInfo('thousands_separator', $form), '')
+                .replace(Give.form.fn.getInfo('decimal_separator', $form), '.');
```
Comment (ravinderk) [inline src/DonationSummary/resources/js/summary.js:215]: @glaubersilva Can you confirm that this logic is compatible with the currency that has `,` as decimal separator like `EUR`

Comment (glaubersilva) [reply src/DonationSummary/resources/js/summary.js]: I did a double check and could notice that the "Donation Total" amount was getting wrong when the value was higher than 999, so I update the normalize amounts logic and removed the extra param added previously.

File: src/DonationSummary/resources/js/summary.js [hunk 2]
```diff
@@ -205,12 +205,13 @@ window.GiveDonationSummary = {
      *
      * @param {string/number} amount
      * @param {jQuery} $form
+     * @param {boolean} normalize
      */
     format_amount: function (amount, $form) {
         // Normalize amounts to JS number format
```
Comment (ravinderk) [inline src/DonationSummary/resources/js/summary.js:211]: @glaubersilva Can you use `Give.fn.unFormatCurrency` and `Give.fn.formatCurrency`. These functions encapsulate current formatting setting logic for a donation form and provide accurate results with add-ons.

**Review by ravinderk — CHANGES_REQUESTED** [review]
@glaubersilva I noticed another issue. If the fee does not display in the donation summary when the admin forces fee optin to the donor.

Screenshot 2022-08-18 at 7 27 17 PM
image

File: src/DonationSummary/resources/js/summary.js (hunk 2 above)
Comment (ravinderk) [reply src/DonationSummary/resources/js/summary.js:211]: @glaubersilva keep in mind that currencies can have ` ` [space], `,`, `.` as thousand separators which can cause unexpected results.

Comment (ravinderk) [reply src/DonationSummary/resources/js/summary.js:210]: fixed: https://github.com/impress-org/givewp/commit/6c17283816f75b0f160591b112c7ac8fd9212450

**Comment by glaubersilva** [comment]
@ravinderk I implemented the requested change and also fixed the format amount args. However, the total donation amount still is getting wrong because of [another issue](https://github.com/impress-org/give-fee-recovery/issues/306) that already has this [other PR](https://github.com/impress-org/give-fee-recovery/pull/307) avaliable.

**Review by ravinderk — APPROVED** [review]

**Comment by jonwaldstein** [comment]
@ravinderk can you please update the PR description to describe the additional changes you made? I see a lot of things were removed and replaced and it's a little hard to understand why those decisions were made by looking at it.

**Comment by glaubersilva** [comment]
@jonwaldstein I updated the PR description with more details as @ravinderk asked me, but if it's not clear enough, let me know.

File: src/DonationSummary/resources/js/summary.js [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
+
     /**
      * @since 2.17.0
      */
     initAmount: function () {
+        const getFormattedDonationAmount = this.getFormattedDonationAmount;
+
         GiveDonationSummary.observe('[name="give-amount"]', function (targetNode, $form) {
-            $form.find('[data-tag="amount"]').html(GiveDonationSummary.format_amount(targetNode.value, $form));
+            $form.find('[data-tag="amount"]').html(getFormattedDonationAmount($form));
```
Comment (jonwaldstein) [inline src/DonationSummary/resources/js/summary.js:37]: Let's remove this local instance `const getFormattedDonationAmount = this.getFormattedDonationAmount;` in favor of the parent method.

```suggestion
            $form.find('[data-tag="amount"]').html(GiveDonationSummary.getFormattedDonationAmount($form));
```

File: src/DonationSummary/resources/js/summary.js [hunk 4]
```diff
@@ -100,50 +118,51 @@ window.GiveDonationSummary = {
...
     initTotal: function () {
+        const handleFees = this.handleFees;
+
         GiveDonationSummary.observe('.give-final-total-amount', function (targetNode, $form) {
-            // Hack: (Currency Switcher) The total is always stored using a the decimal seperator as set by the primary currency.
-            const total = targetNode.dataset.total.replace('.', Give.form.fn.getInfo('decimal_separator', $form));
-            $form.find('[data-tag="total"]').html(GiveDonationSummary.format_amount(total, $form));
+            $form.find('[data-tag="total"]').html(targetNode.textContent);
+
+            handleFees($form);
```
Comment (jonwaldstein) [inline src/DonationSummary/resources/js/summary.js:151]: Same here, we don't need to have a local instance of `const handleFees = this.handleFees;`.

```suggestion
            GiveDonationSummary.handleFees($form);
```

File: src/DonationSummary/resources/js/summary.js [hunk 5]
```diff
@@ -100,50 +118,51 @@ window.GiveDonationSummary = {
...
         });
 
         // Hack: Force an initial mutation for the Total Amount observer
         const totalAmount = document.querySelector('.give-final-total-amount');
         if (totalAmount) {
-            totalAmount.dataset.total = totalAmount.dataset.total;
+            totalAmount.textContent = totalAmount.textContent;
```
Comment (jonwaldstein) [inline src/DonationSummary/resources/js/summary.js:157]: There's a lot of "Hack" comments in this file 😬

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

**Comment by glaubersilva** [comment]
@jonwaldstein I implemented all suggested changes, let me know if there is anything more! ;)

**Review by jonwaldstein — APPROVED** [review]
@glaubersilva thanks for the updates! please send this off to Q/A for final approval

**Review by rickalday — APPROVED** [review]
Tested with Legacy, MSF, and Classic templates.
Working with Donor Opt-in  and Forced Opt-in options too.

**Pull request merged**